specs/lib/combinePrimitivesSpec.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+var fs = require('fs');
+var clone = require('clone');
+var addPipelineExtras = require('../../lib/addPipelineExtras');
+var combinePrimitives = require('../../lib/combinePrimitives');
+var writeGltf = require('../../lib/writeGltf');
+var boxPath = './specs/data/combineObjects/box.gltf';
+var doubleBoxPath = './specs/data/combineObjects/doubleBox.gltf';
+
+describe('addPipelineExtras', function() {
+    var box, doubleBox, doubleBoxError;
+
+    beforeAll(function(done) {
+        fs.readFile(doubleBoxPath, function(err, data) {
+            doubleBox = JSON.parse(data);
+            addPipelineExtras(doubleBox);
+            doubleBoxError = clone(doubleBox);
+
+            fs.readFile(boxPath, function(err, data) {
+                box = JSON.parse(data);
+                done();
+            });
+        });
+    });
+
+    it('does not affect single primitives', function() {
+        var boxCopy = clone(box);
+        addPipelineExtras(boxCopy);    
+        combinePrimitives(boxCopy);
+        writeGltf(boxCopy, './boxOutput.gltf', true, true, function() {
+            expect(boxCopy).toEqual(box);
+        });
+    });
+
+    it('combines two primitives referencing the same bufferView', function() {

```

Comment:
Can we also test the case when primitives can't be combined.  There is some missing coverage: https://coveralls.io/jobs/13867440/source_files/310007864

In general consider:
* When all primitives in a mesh can be combined
* When no primitives in a mesh can be combined
* When some primitives in a mesh can be combined
   * When, for example, a mesh with 5 primitives, can combine 2 primitives into 1, another 2 primitives into 1, and not combine the last primitive so 5 primitives become 3.